Create 6.0.0.md (#465)
silverstripe/developer-docs: en/08_Changelogs/6.0.0.md

---
title: 6.0.0 (unreleased)
---

# 6.0.0 (unreleased)

## Overview

> - [Run CanonicalURLMiddleware in all environment by default](#url-middleware)

Reviewer: ```suggestion
- [Run `CanonicalURLMiddleware` in all environments by default](#url-middleware)
```

- [Bug fixes](#bug-fixes)

> ## Run CanonicalURLMiddleware in all environment by default (#url-middleware)

Reviewer: ```suggestion
## Run `CanonicalURLMiddleware` in all environments by default {#url-middleware}
```

> In Silvestripe 5.0 `CanonicalURLMiddleware` only runs in production by default. This lead to issues with `fetch` and APIs behaving differently in production environments to development. Silverstripe 6.0 changes this default to run the rules in dev, test and live by default.

Reviewer: ```suggestion
In Silverstripe CMS 5 the [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) only runs in production by default. This lead to issues with `fetch` and APIs behaving differently in production environments to development. Silverstripe 6.0 changes this default to run the rules in `dev`, `test`, and `live` by default.
```
Please also include a short yaml snippet showing how to change this (or link to existing docs which show it)

---
title: 6.0.0 (unreleased)
---

# 6.0.0 (unreleased)

## Overview

- [Run `CanonicalURLMiddleware` in all environments by default](#url-middleware)
- [Bug fixes](#bug-fixes)

## Run `CanonicalURLMiddleware` in all environments by default {#url-middleware}

> In Silverstripe CMS 5 [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) only runs in production by default. This lead to issues with `fetch` and APIs behaving differently in production environments to development. Silverstripe 6.0 changes this default to run the rules in dev, test and live by default.

Reviewer: ```suggestion
In Silverstripe CMS 5 [`CanonicalURLMiddleware`](api:SilverStripe\Control\Middleware\CanonicalURLMiddleware) only runs in production by default. This lead to issues with `fetch` and APIs behaving differently in production environments to development. Silverstripe 6.0 changes this default to run the rules in `dev`, `test`, and `live` by default.
```
Originally requested in https://github.com/silverstripe/developer-docs/pull/465#discussion_r1510523381

To opt out of this change include the following in your `_config.php`

```php
use SilverStripe\Core\CoreKernel;

> if (Director::isDev()) {
>     CanonicalURLMiddleware::singleton()->setEnabledEnvs([
>         CoreKernel::LIVE,
>     ]);
> }

Reviewer: ```suggestion
CanonicalURLMiddleware::singleton()->setEnabledEnvs([
    CoreKernel::LIVE,
]);
```
Saying "set it to live when I'm in dev mode" is confusing at best.

Reviewer: Please also add a `use` statement for `CoreKernel`.

Author: @GuySartorelli That's the craziness of the current behaviour!

Reviewer: Oh! Sorry I thought it was the `CoreKernel` use statement that was missing, but it's `CanonicalURLMiddleware`'s use statement that's missing. Can you please add that?
